darmanis_asap_data_prep.Rmd

```diff
@@ -0,0 +1,109 @@
+data <- readr::read_tsv(file.path("darmanis_data", "salmon_quants",
+                                  "darmanis_data_counts.tsv"))
+# Store gene names separately
+gene <- data$gene
+```
+
+### Import metadata
+```{r Set up metadata}
+if (file.exists(file.path("darmanis_data", "meta_data.csv"))) {
+# Read in csv if it's been made, if not read in from GEOquery
+  geo.meta <- read.csv(file.path("darmanis_data", "meta_data.csv"), 
+                       stringsAsFactors = FALSE)
+} else {
+  geo.meta <- GEOquery::getGEO("GSE84465", destdir = "darmanis_data")
+  geo.meta <- data.frame(geo.meta[[1]]@phenoData@data)
+}
+```
+
+### Make sure the samples and the metadata are in the same order.
+```{r Order samples}
+# Put samples in same order 
+id.key <- readRDS(file.path("darmanis_data", "sample_id_key.RDS"))
+
+# Make a sample conversion key
+convert.key <- as.list(as.character(id.key$gsm.ids))
+names(convert.key) <- as.character(id.key$run)
+
+# Obtain GSM ids using conversion key and make these the column names 
+colnames(data) <- dplyr::recode(colnames(data), !!!convert.key)
+
+# Only keep metadata for samples in the set
+geo.meta <- geo.meta[match(colnames(data)[-1], geo.meta$geo_accession), ]
+```
+
+### Make batch file for ASAP to correct batches by.
+In this case we will use the plate ids. 
+ASAP wants the first column to be the sample IDs that match your sample data file
+and the second column to be the corresponding batch the sample belongs to.
+```{r Make batch info}
+# Make batch info matrix
+batch.info <- cbind(geo.meta$geo_accession, geo.meta$plate.id.ch1)
+
+# Let's exclude batches of "1"
+batch.sum <- summary(as.factor(batch.info[,2]))
+batch.one <- names(batch.sum)[which(batch.sum <= 2)]
+
+# Make a regex statement to exclude these batches of one
+regex <- paste0(paste0("^", batch.one, "$"), collapse = "|")
+batch.info <- batch.info[ -grep(regex, batch.info[, 2]), ]
+
+# Exclude these samples from the data
+data <- data %>% dplyr::select(batch.info[,1])
+
+# Look at a preview of what this looks like
+head(batch.info)
+```
+
+### ASAP only likes round numbers for the counts
+Here we have data have it's genes as it's rownames, so even though we are 
+deleting these columns, the information we need is still there.
+Filter out genes that don't have at least one count in 1% of cells and filter out 
+samples that don't express at least 100 genes.
+```{r Filter out genes}
+data <- AsapFilter(data, genes, min_counts = 1, perc_genes = 0.01, num_genes = 100)
+```
+
+### Write data file in the format ASAP will take.
+ASAP wants a typical gene by column gene expression matrix with samples with 
+the column names and the first column being the gene names.
+You can use other forms of separation other than "/t" but will need to specify
+what you have used in a drop down menu on ASAP. 
+```{r Write data file}
+# Write these data to tab delimited text files for use in ASAP
+readr::write_delim(data, file.path("darmanis_data", "asap_format.txt"))
```

Comment:
You use `write_tsv` over here: https://github.com/cansav09/scRNA-seq_workflow/pull/12/files#diff-68147a8e8b26987782b9b5f70e8d65a7R64

Any reason for the distinction?

Reply:
I originally had it as `write_delim`. Both are acceptable to ASAP. However, tsv files are the default, so I changed it to that later. But, I forgot to change it for this dataset